Add task solution

Volodymyr-Bauzi opened the pull request:
[DEMO LINK](https://Volodymyr-Bauzi.github.io/react_dynamic-list-of-goods/)

Volodymyr-Bauzi committed:
Add task solution

S-Khraban commented on src/api/goods.ts:
```diff
@@ -4,13 +4,25 @@ import { Good } from '../types/Good';
 
-export const get5First = () => {
-  return getAll().then(goods => goods); // sort and get the first 5
+export const get5First = (): Promise<Good[]> => {
```
get5First mixed with styles getRedGoods/getAll.
 Don't use numbers, change naming patterns as 
getAllGoods / getFirstFiveGoods / getRedGoods.

S-Khraban commented on src/api/goods.ts:
```diff
@@ -4,13 +4,25 @@ import { Good } from '../types/Good';
-  return getAll().then(goods => goods); // sort and get the first 5
+export const get5First = (): Promise<Good[]> => {
+  return getAll().then(goods => {
+    const sortedGoods = [...goods].sort((a, b) => {
```
I think locale compare is better in this situation 
a.name.localeCompare(b.name, undefined, { sensitivity: 'base' }).

S-Khraban commented on src/api/goods.ts:
```diff
@@ -4,13 +4,25 @@ import { Good } from '../types/Good';
+    });
+
+    return sortedGoods.slice(0, 5);
+  }); // sort and get the first 5
```
Delete comments here

S-Khraban commented on src/api/goods.ts:
```diff
@@ -4,13 +4,25 @@ import { Good } from '../types/Good';
-export const getRedGoods = () => {
-  return getAll().then(goods => goods); // get only red
+export const getRedGoods = (): Promise<Good[]> => {
+  return getAll().then(goods => goods.filter(good => good.color === 'red')); // get only red
```
and here

S-Khraban commented on src/App.tsx:
```diff
@@ -1,27 +1,70 @@
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 // or
 // import * as goodsAPI from './api/goods';
```
Use classnames

S-Khraban commented on src/App.tsx:
```diff
@@ -1,27 +1,70 @@
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
```
I wrote about naming in prev one/

S-Khraban commented on src/GoodsList.tsx:
```diff
@@ -1,16 +1,18 @@
 };
 
-export const GoodsList: React.FC<Props> = ({ goods }) => (
+export const GoodsList: React.FC<Props> = memo(({ goods }) => (
```
ChatGPT сказал:

memo returns a MemoExoticComponent<...>, so assigning it to a variable typed as React.FC<Props> is incorrect.
Declare the component separately and then wrap it with memo, or use the generic version of memo.

S-Khraban reviewed (changes requested):
Good work, but there are some problematic areas that need to be changed.

Volodymyr-Bauzi committed:
fix:
- change naming patterns
- fixed sorting a little with adding ignore sensitivity
- deleted comments
- corrected memo usage

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Excellent work on this project! I'm approving your solution as it successfully meets all the requirements. Your code is clean, well-structured, and demonstrates a strong understanding of React principles. I was particularly impressed with the clear separation of concerns, with data fetching and preparation logic correctly isolated in `src/api/goods.ts`. You've also correctly implemented key React optimizations like `React.memo` and list `key` props in the `GoodsList` component. Great job!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)
